test/object_store/run

```diff
@@ -86,27 +86,29 @@ def run_with_dir(run_cmd_gen, run_dir):
     return pid
 
 
-def restart_with_dir(old_pid, run_cmd_gen, run_dir):
+def kill_with_dir(old_pid, run_dir):
```

Comment:
This looks like a "fix" to patch 6261bba366d5c504eaf96204e1ed01a011d607b6 from this set. Why not make it look like here from the very beginning?

Reply:
as the commit message explains, i want to make smaller steps to ease the review. 6261bba366d5c504eaf96204e1ed01a011d607b6 is only supposed to copy the vendored method with minimal editings. instead of fixing them.